src/main/java/org/opensearch/ad/model/AnomalyDetector.java

```diff
@@ -720,4 +723,117 @@ private static Boolean onlyParseBooleanValue(XContentParser parser) throws IOExc
         }
         return null;
     }
+
+    /**
+     * Validates each condition in the list of rules against the list of features.
+     * Checks that:
+     * - The feature name exists in the list of features.
+     * - The related feature is enabled.
+     * - The value is not NaN and is positive.
+     *
+     * @param features The list of available features. Must not be null.
+     * @param rules The list of rules containing conditions to validate. Can be null.
+     */
+    private void validateRules(List<Feature> features, List<Rule> rules) {
+        // Null check for rules
+        if (rules == null || rules.isEmpty()) {
+            return; // No suppression rules to validate; consider as valid
+        }
+
+        // Null check for features
+        if (features == null) {
+            // Cannot proceed with validation if features are null but rules are not null
```

Comment:
nit: could add `|| features.isEmpty()` same we do for rules above

Reply:
yeah, fixed.